### PR TITLE
gruvbox-dark-gtk: init at 1.0.1, gruvbox-dark-icons-gtk: init at 1.0.0

### PR DESCRIPTION
**Pull request by NomisIV**
###### Motivation for this change
I use these packages locally, and wanted them to become official. I have got them working properly, but the icon preview in lxappearance does not work. I don't know if it's a problem with the packaging, or the project. Icons seem to work nonetheless.

Please bear in mind that I'm new to NixOS, and contributing generally, so I might have missed something

Here are the git repositories for the projects:
https://github.com/jmattheis/gruvbox-dark-gtk
https://github.com/jmattheis/gruvbox-dark-icons-gtk

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`(completed with `No diff detected, stopping review...`)
- [x] Tested execution of all binary files (usually in `./result/bin/`) (The packages successfully build, and the themes can be applied using lxappearance)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  '';
+
+  dontDropIconThemeCache = true;
+
+  meta = with stdenv.lib; {
+    description = "Gruvbox icons for GTK based desktop environments";
+    homepage = "https://github.com/jmattheis/gruvbox-dark-gtk";
+    license = licenses.gpl3;
+    platforms = platforms.unix;
+    manitainers = [ maintainers.nomisiv ];
```
Comment (cole-h) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:27]: You need to add yourself as a maintainer in the [maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/c44acaaceb04ab149b805656eeffcdf2acf939e6/maintainers/maintainer-list.nix) file for this to work.

Please do so in a separate commit (`maintainers: add nomisiv`).

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub }:
```
Comment (cole-h) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub }:
```

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/icons/
+    rm README.md
+    cp -r ${src} $out/share/icons/${pname}
+  '';
+
+  dontDropIconThemeCache = true;
+
+  meta = with stdenv.lib; {
```
Comment (cole-h) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:22]:
```suggestion
  meta = with lib; {
```

File: pkgs/data/themes/gruvbox-dark-gtk/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub }:
```
Comment (cole-h) [inline pkgs/data/themes/gruvbox-dark-gtk/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub }:
```

File: pkgs/data/themes/gruvbox-dark-gtk/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "v${version}";
+    sha256 = "1wf4ybnjdp2kbbvz7pmqdnzk94axaqx5ws18f34hrg4y267n0w4g";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/themes/
+    cp -r ${src} $out/share/themes/${pname}
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (cole-h) [inline pkgs/data/themes/gruvbox-dark-gtk/default.nix:19]:
```suggestion
  meta = with lib; {
```

**Review by cole-h — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
I just renamed the title to more easily reflect the commits that are in here.

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    description = "Gruvbox icons for GTK based desktop environments";
+    homepage = "https://github.com/jmattheis/gruvbox-dark-gtk";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:25]: gpl3Plus or gpl3Only?

Comment (NomisIV) [reply pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:25]: gpl3Only, I think? Is licenses.gpl3 is deprecated, or is licenses.gpl3Olny simply preferred?
Link: https://github.com/jmattheis/gruvbox-dark-gtk/blob/master/LICENSE

File: pkgs/data/themes/gruvbox-dark-gtk/default.nix (hunk 5 above)
Comment (NomisIV) [reply pkgs/data/themes/gruvbox-dark-gtk/default.nix:19]: Is there any particular reason this method is preferred? They seem equivalent to me, but I might be missing something

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix (hunk 6 above)
Comment (cole-h) [reply pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:25]: `licenses.gpl3` is deprecated indeed.

https://github.com/NixOS/nixpkgs/blob/bc91fe1935b7ac1699e166bd99863a0ae6b6efde/lib/licenses.nix#L850-L853

File: pkgs/data/themes/gruvbox-dark-gtk/default.nix (hunk 5 above)
Comment (cole-h) [reply pkgs/data/themes/gruvbox-dark-gtk/default.nix:19]: https://github.com/NixOS/nixpkgs/issues/108938

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix (hunk 1 above)
Comment (cole-h) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:27]:
```suggestion
    maintainers = [ maintainers.nomisiv ];
```

Comment (NomisIV) [reply pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:27]: Oh, silly me 😂

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash the package related commits together.

**Comment by NomisIV** [comment]
Well, that didn't work like intended...

**Comment by SuperSandro2000** [comment]
The right command is head ``git rebase -i @~9``. I am not sure what you did now but it is not the right thing and merging master into a PR is a big no go.

**Comment by cole-h** [comment]
(FWIW, `HEAD~9` works fine.)

You first do the git rebase command you used. Then, you choose the commits you want to squash, but you must order them to come DIRECTLY AFTER the commit you want to squash them into. e.g.

```
pick asdf1 commit 1
pick asdf2 commit 2
pick asdf3 commit 3, intended to become part of commit 1
pick asdf4 commit 4, intended to become part of commit 2
pick asdf5 commit 5, intended to become part of commit 1
pick asdf6 commit 6, intended to become part of commit 1
pick asdf7 commit 7, intended to become part of commit 2
```

would become

```
pick asdf1 commit 1
squash asdf3 commit 3, intended to become part of commit 1
squash asdf5 commit 5, intended to become part of commit 1
squash asdf6 commit 6, intended to become part of commit 1
pick asdf2 commit 2
squash asdf4 commit 4, intended to become part of commit 2
squash asdf7 commit 7, intended to become part of commit 2
```

I usually use `fixup` instead of `squash`, since I don't care about the commit message most times.

**Comment by SuperSandro2000** [comment]
> I usually use fixup instead of squash, since I don't care about the commit message most times.

Actually this is what I do all the time, too but it is still called squashing.

**Comment by NomisIV** [comment]
> but you must order them to come DIRECTLY AFTER the commit you want to squash them into

That's exactly what I did, so I don't know what I did wrong.

> I am not sure what you did now but it is not the right thing and merging master into a PR is a big no go.

Are you referring to this pull request [971f2a1](https://github.com/NixOS/nixpkgs/pull/109585/commits/971f2a1ca022921e3066daa19809453ed64fc917)? If yes, I don't know how to avoid doing that. After the rebase git wanted me to pull changes for some reason, and since I had local unpushed commits, the merge emerged.

Is there a simple way to rescue this disaster? I'm thinking of attempting something like [this](https://gist.github.com/dsci/1347672/da3dbda654fb0947aec4a43218bd64c2acdf7871), but figured I should probably ask someone who knows what to do before I screw things up more.

Another thing I could try is to start from scratch and redo the changes in the PR, but I'd like to learn how to do it properly too.

I learn from my mistakes ;)

**Comment by SuperSandro2000** [comment]
You can take the changes from this PR, delete the branch, create it under the new name and commit the changes into 3 commits and then force push.

**Comment by cole-h** [comment]
Instead of creating a merge commit, you usually want to `git pull --rebase` which will automatically rebase your local changes on top of wherever you're pulling from. If there are conflicts, you'll be able to fix them and `git rebase --continue` until your changes are "compatible" with wherever you pulled from.

File: pkgs/data/themes/gruvbox-dark-gtk/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "jmattheis";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wf4ybnjdp2kbbvz7pmqdnzk94axaqx5ws18f34hrg4y267n0w4g";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/themes/
+    cp -r ${src} $out/share/themes/${pname}
```
Comment (romildo) [inline pkgs/data/themes/gruvbox-dark-gtk/default.nix:16]: In the distributed `index.theme` file, the `Name` field is set to `gruvbox-dark`, so it would be better to install to `$out/share/themes/gruvbox-dark`.

Maybe it is better not to install the `.github` subdirectory and the `LICENSE` and `README.md` files.

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "gruvbox-dark-icons-gtk";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "jmattheis";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1fks2rrrb62ybzn8gqan5swcgksrb579vk37bx4xpwkc552dz2z2";
+  };
+
```
Comment (romildo) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:13]: Add the new dependencies.

```suggestion

  nativeBuildInputs = [ gtk3 ];

  propagatedBuildInputs = [ breeze-icons gnome-icon-theme hicolor-icon-theme ];
```

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  installPhase = ''
+    mkdir -p $out/share/icons/
```
Comment (romildo) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:15]: Install to `oomox-gruvbox-dark`, which is the name of the icon theme according to the `index.theme` file.
```suggestion
    mkdir -p $out/share/icons/oomox-gruvbox-dark
```

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    rm README.md
+    cp -r ${src} $out/share/icons/${pname}
```
Comment (romildo) [inline pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix:17]: Do not copy `${src}` directly, because it is read only and it will be needed write  permission in order to generate the cache.

```suggestion
    cp -r * $out/share/icons/oomox-gruvbox-dark
    gtk-update-icon-cache $out/share/icons/oomox-gruvbox-dark
```

**Review by romildo — CHANGES_REQUESTED** [review]

**Comment by NomisIV** [comment]
I was able to fix the problem by rebasing a second time, and using `git push --force` to push the changes. I didn't use the force flag the first time, and that's probably why the commits didn't squash.

File: pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix (hunk 10 above)
Comment (NomisIV) [reply pkgs/data/icons/gruvbox-dark-icons-gtk/default.nix]: Is the dollar sign in `$out` not needed?

**Comment by NomisIV** [comment]
I have now added @romildo's suggested changes, some even in both files (like removing unneeded files, and not copying `${src}` directly)

File: pkgs/top-level/all-packages.nix [hunk 11]
```diff
@@ -20322,6 +20322,10 @@ in
 
   greybird = callPackage ../data/themes/greybird { };
 
+  gruvbox-dark-gtk = callPackage ../data/themes/gruvbox-dark-gtk { };
+
+  gruvbox-dark-icons-gtk = callPackage ../data/icons/gruvbox-dark-icons-gtk { };
```
Comment (cole-h) [inline pkgs/top-level/all-packages.nix:20327]:
```suggestion
  gruvbox-dark-icons-gtk = callPackage ../data/icons/gruvbox-dark-icons-gtk {
    inherit (plasma5Packages) breeze-icons;
  };
```

Comment (NomisIV) [reply pkgs/top-level/all-packages.nix:20327]: I was getting an error saying `error: undefined variable 'plasma5Packages' at ...`

I tried looking around the file and found that the line other packages used was `inherit (kdeFrameworks) breeze-icons`. I tried using that instead, and it works.

**Review by cole-h — APPROVED** [review]
`kdeFrameworks` doesn't appear in `nix repl .`, which is probably why eval failed. I changed `kdeFrameworks` to `plasma5Packages`, reordered the commits, and force-pushed, which fixed the eval error.

Now, diff LGTM. GitHub ID also checks out. Thanks :)
